flask/main.py

```diff
@@ -114,7 +119,12 @@ def products_join():
         raise(err)
 
     try:
-        r = requests.get(RUBY_BACKEND + "/api")
+        headers = {
+            "se" : None if request.headers.get('Se') == 'undefined' else request.headers.get('Se'),
+            "customerType" : None if request.headers.get('Customertype') == 'undefined' else request.headers.get('Customertype'),
+            "email" : None if request.headers.get('Email') == 'undefined' else request.headers.get('Email')
+        }
+        r = requests.get(RUBY_BACKEND + "/api", headers=headers)
```

Comment:
Alternatively, we could add a method on `utils.py` that parses the headers and call it from here, since this same code is repeated twice.

Reply:
@sergiosentry sounds good, yes please do that with `urils.py`

One question.

How did you settle on checking the string as `.get('Customertype')` versus `.get('customerType')`? We try to write it as customerType, coming from the client side (web JS, mobile), and then check for it as 'customerType' in our server api (here's [how Flask does it](https://github.com/sentry-demos/application-monitoring/blob/a125ed8dd3a448c949b0e90e506434ba9c48c119/flask/main.py#L156), as an example), but maybe ruby force lowercasing it then capitalizing the first letter? Is that what you found?

I guess it's possible that Express, Spring and Flask all force lowercase it as they send the request outbound to Ruby? I feel like I've seen the casing of request header strings get adjusted before, bot not sure.

Did you re-test? Could you update the transaction link showing that it's working for React<>Flask<>Ruby?

Reply:
@thinkocapo Got it!
Mhmm weird, I can get the values of the headers by using both `customerType` and `Customertype`. I guess is not case-sensitive for the `request.headers` object in Flask? I do know that Ruby converts headers to uppercase by convention [reference](https://jatindhankhar.in/blog/custom-http-header-and-ruby-standard-library/#:~:text=Since%20for%20most%20of%20the,of%20the%20third%20party%20apis.) 

Anyways, I changed it to use the same casing as the react frontend (`customerType`) and it works! I also updated the transaction link